prow/prowjobs/GoogleContainerTools/kpt-config-sync/kpt-config-sync-periodics.yaml

```diff
@@ -327,7 +327,27 @@ periodics:
       - 'GKE_RELEASE_CHANNEL=regular'
       - 'E2E_CLUSTER_PREFIX=standard-regular-gcenode'
       - 'E2E_NUM_CLUSTERS=1'
-      - 'E2E_ARGS=--gcenode -run=TestGCENode'
+      - 'E2E_GIT_PROVIDER=csr'
+      - 'E2E_ARGS=--gcenode -run=TestGCENode*'
+
+- <<: *config-sync-standard-job
+  name: kpt-config-sync-standard-regular-workloadidentity
+  annotations:
+    testgrid-dashboards: googleoss-kpt-config-sync-main
+    testgrid-tab-name: standard-regular-workloadidentity
+  spec:
+    <<: *config-sync-standard-job-spec
+    containers:
+    - <<: *config-sync-standard-job-container
+      args:
+      - 'GCP_SUBNETWORK=prow-e2e-subnetwork-usw1-11'
+      - 'GKE_RELEASE_CHANNEL=regular'
+      - 'E2E_CLUSTER_PREFIX=standard-regular-workloadidentity'
+      - 'E2E_NUM_CLUSTERS=1'
+      - 'E2E_GIT_PROVIDER=csr'  # run CSR+WI tests on a separate job
```

Comment:
Should we consider defaulting to `csr` instead of `local` for the other jobs?

Reply:
I don't think all tests in other jobs can run with CSR, for example, proxy test, ca_cert test, and etc.
If they have to run with the local test-git-server, we should default to `local`.

Reply:
It looks like it's only the `caCert` and `knownHosts` tests which require local git provider, which adds up to 5 tests. Either way we are skipping some tests, and note that the kind cluster tests (both presubmit and periodic) will run with `local` git provider to exercise `caCert`/`knownHosts`. I also feel that test coverage for CSR may be more meaningful than our bespoke local git provider.

We discussed this offline, but it still feels like a bit of an inconsistency that we run the full test suite for gitlab/bitbucket but only this small subset of coverage for CSR.

Let me know what you think, I'm fine with this if you disagree

Reply:
I like the idea of having more test coverage for CSR. If we make CSR the default git-server, then we don't need a separate job to run the GIT+WI tests.
My two concerns are:
1. It may take longer for each test to provision a  CSR repo and push commits than using the local test-git-server, but I'm not sure how much longer it is. 
2. We need to add a `ntopts.RequireLocalGitServer` to filter out those tests that require `local` git provider,  including `no_ssl_verify_test.go`, `private_cert_secret_test.go`, `proxy_test.go`, and `ssh_known_hosts_test.go`. I'm not too concerned since they're covered by the kind cluster tests.

That being said, it may be worth to exercise the change and check the overall test duration.

Reply:
Nevermind, `private_cert_secret_test.go` and `ssh_known_hosts_test.go` already have `ntopts.RequireLocalGitProvider`.
`proxy_test.go` sets up the http proxy with a GitHub repo, and `no_ssl_verify_test.go` verifies if the env var is configured.

This PR is updated to change the default git provider to CSR.